### DISCUSSION TITLE
What are reporting consumed sensors?

### DISCUSSION THREAD

**Discussion by rzulian** [gekkekoe/esphome-ecodan-hp#35] [General]
What are consumed sensors reporting? It's not really clear. 
I though it was the daily consumed, but it doesn't look so. Moreover they are not changing at midnight, is this due to the time saving period?

 
<img width="429" alt="Screenshot 2024-08-03 alle 07 09 11" src="https://github.com/user-attachments/assets/88803cc8-db35-4feb-8fc9-21851a73b9d5">

**Comment by gekkekoe** [comment]
Previous daily consumed kWh (heating/cool/DHW). 

it is reported around midnight. you probably have the time (summer time switch) set incorrectly.

btw, questions can be asked in the discussions section. will move this there eventually.

**Comment by rzulian** [comment]
Do you mean that  is incorrect in the ecodan or in the component?
Do I need an utility meter to get it correctly, assuming the time is correct?

**Comment by gekkekoe** [comment]
setting in ecodan that has to be manually switched each season...

for meters please read  https://github.com/gekkekoe/esphome-ecodan-hp/discussions/22

**Reply by rzulian** [reply]
I've tried to follow the instructions regarding the utility meters, but I think there is something wrong.
You are suggesting to use delta values = true, but the instructions are 
`Delta values
Enable this if the source values are delta values since the last reading instead of absolute values. When this option is enabled, each new value received will be added as-is to the utility meter instead of adding the difference between the new value and previous value.`
So the consumed sensors are already "daily utility meters", but the problem is that they are referring to the previous day. Is there a way to solve this? How are you getting these values in the energy dashboard?

**Comment by gekkekoe** [comment]
I have a homewizard energy meter, so I'm not using the reported numbers other than just an indication (they are far from accurate compared to the homewizard as explained in the other thread).

I haven't used the utility meters myself, but you should be able to **read it daily**. You might want to consult the home assistant forum, there's a melcloud integration thread, they are using the same reported values, and probably someone already did the work...

**Reply by rzulian** [reply]
How do you distinguish between DWH and cooling/heating using an external energy meter? Do you have a template enrgy meter based on the ecodan status?

Another question: is it available the total consumed energy from the beginning (always increasing) ? That would be perfect for the energy dashboard.

**Reply by gekkekoe** [reply]
I actually do not really care about the breakdown of the consumption. But you could make something based on the operation status via template sensors as you suggested. For me heating and cooling is the majority of the consumption, dhw is only small portion. I can look at the ecodan numbers to get an indication of the breakdown if I wanted.

For data that is available over cn105, please have a look at https://www.logicool-ac.com/wp-content/uploads/2018/08/MELCOBEMS-MINI-A1M-ATW-Modbus-Register-Tables.pdf and see whats available for FTC4, 5, 6. But given that it's not accurate, it's not really worthwhile to reverse engineer these. If you wire a pulse energy meter to the ecodan, then the number becomes accurate, but probably not a lot of ppl are doing this.

**Comment by gekkekoe** [comment]
put this file in same folder as configuration.yaml
https://github.com/gekkekoe/esphome-ecodan-hp/blob/main/automations/energy.yaml
add entry in the config file:
`utility_meter: !include energy.yaml`
reboot, the values will accumulate on change, and will cycle monthly. You won't see anything until the first change.

**Reply by rzulian** [reply]
@gekkekoe in your experience what's the difference between the ecodan estimated energy and the energy from a pulse meter?

**Reply by gekkekoe** [reply]
looking at last 2 months, it was off between 15%-20% for the whole month

**Reply by rzulian** [reply]
I’m getting a 50% more from the energy meter vs the ecodan estimated. Now I need something else to determine which is right

**Reply by gekkekoe** [reply]
my meter is certified, so pretty sure it's correct. but the ecodan internal metering is off, and only become accurate when a pulse meter is wired into it

**Reply by rzulian** [reply]
How do you add a pulse meter to the ecodan?

**Reply by F1p** [reply]
My Ecodan Consumed vs. Energy Meter is +/-5% accurate on Hot Water
But the Ecodan is estimating average of +27% higher consumed energy than actual on Heating

**Reply by gekkekoe** [reply]
@F1p  yeah it's a joke and I stopped looking at those numbers.
@rzulian please consult the service manual, there is a connection on the inside unit to wire a pulse meter to it. But it still remains a bit difficult to accumulate the totals due to they way they report it. So advice would be: get a smart meter and monitor it outside the ecodan. then you can compute realtime COP based on estimated power output and actual power usage. you can add that meter also the the energy dashboard of HA. It's a much cleaner solution. (for me I actually don't care about the ratio heating/hotwater/cooling/standby), I just want the total usage. Keep in mind that the booster element is a separate group

If you have a meter, then you also see funny things, like around 30W standby usage with peaks to 70W. Its quite high compared to other heatpumps

**Reply by rzulian** [reply]
I've added the Vue monitoring to the Ecodan, and it's reporting a huge difference from the estimated (eg. Vue 20 kWh vs 12 Kwh )!
Is that possible ?!?

**Reply by gekkekoe** [reply]
that seems a bit excessive, for me around 10-20%. Are you measuring the correct circuits? Also, the booster heater is not included on the main heatpump circuit.

**Reply by F1p** [reply]
> that seems a bit excessive, for me around 10-20%. Are you measuring the correct circuits? Also, the booster heater is not included on the main heatpump circuit.

But booster energy is also not included in consumption/delivery estimate

**Reply by gekkekoe** [reply]
no but he might be measuring it with Vue and compare it with the ecodan numbers.

**Reply by rzulian** [reply]
Yes, I'm measuring it using the Emporia Vue. I've installed the three CTs after the breaker for the Ecodan unit, so I should get all the energy.
Maybe I need a third system to check who is right?

**Reply by gekkekoe** [reply]
looks wrong probably, do you also have a total consumption meter? you can compare if the Vue is reporting more than the total consumption when using the heatpump.

**Comment by F1p** [comment]
@gekkekoe - I also had a 50W standby for the outdoor unit, but it's not accounted for in the estimated consumed figures - i think it only starts measuring when in an operation mode.

**Reply by gekkekoe** [reply]
you can check this in month that you do not use cooling/heating and the active mode is cooling/heating. If it reports 0 when you did not use it, then it does not account for it.
